compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -271,7 +271,7 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
     }
 
     fn normalize(mut self) -> Vec<traits::PredicateObligation<'tcx>> {
-        let cause = self.cause(traits::MiscObligation);
+        let cause = self.cause(traits::WellFormed(None));
```

Comment:
this commit has no effect at all? Or did you forget to update tests?

Reply:
None of the `WellFormed` changes have any output impact, but I want to reduce the number of `MiscObligation`s being used. I find them to make things harder to debug way too often.